cmd/install/install.go

```diff
@@ -0,0 +1,233 @@
+			fmt.Println(" → Error installing the latest kubectl version")
+			return err
+		}
+		_, err = exec.Command("chmod", "+x", kubectlPath).Output()
+		if err != nil {
+			fmt.Println(" → Error giving access to kubectl")
+			return err
+		}
+
+	case "windows":
+		_, err := exec.LookPath("curl")
+		if err != nil {
+			return ErrCurlMissing
+		}
+		err = downloadKubectlLatest(osPlatform)
+		if err != nil {
+			fmt.Println(" → Error installing the latest kubectl version")
+			return err
+		}
+		fmt.Println(" → Add kubectl binary to your windows path")
+
+	case "linux":
+		err = downloadKubectlLatest(osPlatform)
+		if err != nil {
+			fmt.Println(" → Error installing the latest kubectl version")
+			return err
+		}
+		_, err = exec.Command("chmod", "+x", kubectlPath).Output()
+		if err != nil {
+			fmt.Println(" → Error giving access to kubectl")
+			return err
+		}
+	}
+	return nil
+}
+
+// Downloads the latest version of kubectl
+func downloadKubectlLatest(osPlatform string) error {
+	latestVersion := "https://storage.googleapis.com/kubernetes-release/release/stable.txt"
+	var err error
+	ver, err := exec.Command("curl", "-sL", latestVersion).Output()
+	if err != nil {
+		return err
+	}
+	var downloadKubectl string
+	if osPlatform == "windows" {
+		downloadKubectl = "https://storage.googleapis.com/kubernetes-release/release/" + string(ver) + "/bin/" + osPlatform + "/amd64/kubectl.exe"
+	} else {
+		downloadKubectl = "https://storage.googleapis.com/kubernetes-release/release/" + string(ver) + "/bin/" + osPlatform + "/amd64/kubectl"
+	}
+	_, err = exec.Command("curl", "-o", kubectlPath, "-LO", downloadKubectl).Output()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Check the platform and on the basis of that install argocd
+func installArgoCDPlatform() error {
+	var err error
+	osPlatform := runtime.GOOS
+	fmt.Println(" → Installing argocd")
+	switch osPlatform {
+	case "darwin":
+		err = downloadArgoCDLatest(osPlatform)
+		if err != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		_, err = exec.Command("chmod", "+x", argocdPath).Output()
+		if err != nil {
+			fmt.Println(" → Error giving access to argocd")
+			return err
+		}
+
+	case "windows":
+		_, err := exec.LookPath("curl")
+		if err != nil {
+			return ErrCurlMissing
+		}
+		err = downloadArgoCDLatest(osPlatform)
+		if err != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		fmt.Println(" → Add argocd binary to your windows path")
+
+	case "linux":
+		err = downloadArgoCDLatest(osPlatform)
+		if err != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		_, err = exec.Command("chmod", "+x", argocdPath).Output()
+		if err != nil {
+			fmt.Println(" → Error giving execute permission to argocd (ArgoCD CLI)")
+			return err
+		}
+	}
+	return nil
+}
+
+// Downloads the latest version of argocd
+func downloadArgoCDLatest(osPlatform string) error {
+	var downloadArgoCD string
+	argocdVersion := "v2.2.12"
```

Comment:
Not the latest, and should probably make it configurable / overridable, perhaps via environment variable.

Reply:
Currently I have set the argocdVersion variable to v2.2.12, but this will need to change as the version we support changes.  I'll look into how we can make this configurable .